AzureDevOpsPipelines/deployment/bot-publish/source/Core.ts

```diff
@@ -0,0 +1,110 @@
+            let cliPassword: string;
+            var servicePrincipalId: string = tl.getEndpointAuthorizationParameter(connectedService, "serviceprincipalid", false);
+            if (authType == "spnCertificate") {
+                tl.debug('certificate based endpoint');
+                let certificateContent: string = tl.getEndpointAuthorizationParameter(connectedService, "servicePrincipalCertificate", false);
+                cliPassword = path.join(tl.getVariable('Agent.TempDirectory') || tl.getVariable('system.DefaultWorkingDirectory'), 'spnCert.pem');
+                fs.writeFileSync(cliPassword, certificateContent);
+                this.cliPasswordPath = cliPassword;
+
+            }
+            else {
+                tl.debug('key based endpoint');
+                cliPassword = tl.getEndpointAuthorizationParameter(connectedService, "serviceprincipalkey", false);
+                this.servicePrincipalId = servicePrincipalId;
+                this.servicePrincipalKey = cliPassword;
+            }
+            var tenantId: string = tl.getEndpointAuthorizationParameter(connectedService, "tenantid", false);
+
+            //login using svn
+            this.throwIfError(tl.execSync("az", `login --service-principal -u "${servicePrincipalId}" -p "${cliPassword}" --tenant "${tenantId}"`), "Azure Login Failed!");
+        }
+        else if(authScheme.toLowerCase() == "managedserviceidentity") {
+            //login using msi
+            this.throwIfError(tl.execSync("az", "login --identity"), "Managed Service Identity Failed!");
+        }
+        else{
+            throw console.error("Auth Scheme %s is not supported", authScheme);
+        }
+
+        this.isLoggedIn = true;
+        //set the subscription imported to the current subscription
+        this.throwIfError(tl.execSync("az", `account set --subscription "${this.subscriptionID}"`), "Error in setting up subscription");
+    }
+
+    private throwIfError(resultOfToolExecution: IExecSyncResult, errormsg?: string): void {
+        if (resultOfToolExecution.code != 0) {
+            tl.error("Error Code: [" + resultOfToolExecution.code + "]");
+            if (errormsg) {
+                tl.error("Error: " + errormsg);
+            }
+            throw resultOfToolExecution;
+        }
+    }
+
+    private getCWD(cwd: string): IExecOptions {
+
+        return <IExecOptions> { cwd: tl.getInput(cwd) };
+    }
+
+    public GetParameters(parameters: any): string {
+        var inputs: string[] = [];
+        for (var key in parameters) {
+            var input = tl.getInput(key);
+            if (input != null) {
+                inputs.push(`--${key} ${input}`);
+            }
+        }
+        
+        return inputs.join(' ');
+    }
+
+    public run(command: string, options?: IExecOptions): void{
+        var command = `${this.prefix} ${command}`;
+        console.log(`Running command: ${command}.`);
```

Comment:
We'd avoid console logs. Is there any reason to keep them?

Reply:
Actually yes, we need log to the Pipeline log what is being executed.